flink-kubernetes-operator-api/src/main/java/org/apache/flink/kubernetes/operator/api/status/FlinkDeploymentStatus.java

```diff
@@ -55,4 +60,175 @@ public class FlinkDeploymentStatus extends CommonStatus<FlinkDeploymentSpec> {
+                            conditions,
+                            ConditionUtils.runningFalse(
+                                    "JobManager deployment not found",
+                                    "JobManager deployment not found"));
+                    break;
+                case DEPLOYING:
+                    updateCondition(
+                            conditions,
+                            ConditionUtils.runningFalse(
+                                    "JobManager process is starting up",
+                                    "JobManager process is starting up"));
+                    break;
+                case DEPLOYED_NOT_READY:
+                    updateCondition(
+                            conditions,
+                            ConditionUtils.runningFalse(
+                                    "JobManager is running but not ready yet to receive REST API calls",
+                                    "JobManager is running but not ready yet to receive REST API calls"));
+                    break;
+                case ERROR:
+                    updateCondition(
+                            conditions,
+                            ConditionUtils.runningFalse(
+                                    "Deployment in terminal error, requires spec change for reconciliation to continue",
+                                    "JobManager deployment failed"));
+            }
+        } else if (getJobStatus() != null && getJobStatus().getState() != null) {
+            switch (getJobStatus().getState()) {
+                case RECONCILING:
+                    updateCondition(
+                            conditions,
+                            ConditionUtils.runningFalse(
+                                    JobStatus.RECONCILING.name(), "Job is currently reconciling"));
+                    break;
+                case CREATED:
+                    updateCondition(
+                            conditions,
+                            ConditionUtils.runningFalse(
+                                    JobStatus.CREATED.name(), "Job is created"));
+                    break;
+                case RUNNING:
+                    updateCondition(
+                            conditions,
+                            ConditionUtils.runningTrue(JobStatus.RUNNING.name(), "Job is running"));
+                    break;
+                case FAILING:
+                    updateCondition(
+                            conditions,
+                            ConditionUtils.runningFalse(
+                                    JobStatus.FAILING.name(), "Job has failed"));
+                    break;
+                case RESTARTING:
+                    updateCondition(
+                            conditions,
+                            ConditionUtils.runningFalse(
+                                    JobStatus.RESTARTING.name(),
+                                    "The job is currently undergoing a restarting"));
```

Comment:
nit:  "The job is currently undergoing a restarting" ->  "The job is currently restarting"